### PR TITLE
epoll: introduce test-open and improve logging on failure in xapi startup and database. code

### PR DESCRIPTION
**Pull request by edwintorok**
I think these are the only changes I have before we'd attempt a merge of feature/perf (with more epoll related content that I'll move to feature/epoll instead).

### CODE REVIEW

File: ocaml/xapi/xapi.ml [hunk 1]
```diff
@@ -930,12 +930,21 @@ let report_tls_verification ~__context =
   let value = Stunnel_client.get_verify_by_default () in
   Db.Host.set_tls_verification_enabled ~__context ~self ~value
 
+let test_open () =
```
Comment (lindig) [inline ocaml/xapi/xapi.ml:933]: I would have passed the `count` as argument and used `__FUNCTION__` as part of the log message. I assume 
`Xapi_stdext_unix.Unixext.test_open` can fail. Is this properly handled?

**Review by lindig — APPROVED** [review]

**Review by psafont — APPROVED** [review]

File: ocaml/xapi/xapi.ml (hunk 1 above)
Comment (edwintorok) [reply ocaml/xapi/xapi.ml]: Good point, the implementation of `test_open` is completely missing from this PR, so we can't see what it does (it is in the other PR), I probably forgot to cherry-pick it, fixed that now.

If `test_open` fails then XAPI itself should fail to start up with a proper error message. In theory. I'll test what actually happens and report back.

Comment (edwintorok) [reply ocaml/xapi/xapi.ml]: If `test-open` fails then XAPI will fail to startup with this message:
```

Comment (edwintorok) [reply ocaml/xapi/xapi.ml]: Looking better now:
```
 ```
    Jun 19 04:58:41 lcy2-dt72 xapi: [error||0 ||backtrace] Xapi.watchdog failed with exception Unix.Unix_error(Unix.EMFILE, "dup", "")
    Jun 19 04:58:41 lcy2-dt72 xapi: [error||0 ||backtrace] Raised Unix.Unix_error(Unix.EMFILE, "dup", "")
    Jun 19 04:58:41 lcy2-dt72 xapi: [error||0 ||backtrace] 1/7 xapi Raised at file ocaml/libs/xapi-stdext/lib/xapi-stdext-unix/unixext.ml, line 873
    Jun 19 04:58:41 lcy2-dt72 xapi: [error||0 ||backtrace] 2/7 xapi Called from file ocaml/xapi/xapi.ml, line 936
    Jun 19 04:58:41 lcy2-dt72 xapi: [error||0 ||backtrace] 3/7 xapi Called from file ocaml/xapi/xapi.ml, line 946
    Jun 19 04:58:41 lcy2-dt72 xapi: [error||0 ||backtrace] 4/7 xapi Called from file ocaml/xapi/xapi.ml, line 1535
    Jun 19 04:58:41 lcy2-dt72 xapi: [error||0 ||backtrace] 5/7 xapi Called from file ocaml/xapi/xapi.ml, line 1541
    Jun 19 04:58:41 lcy2-dt72 xapi: [error||0 ||backtrace] 6/7 xapi Called from file ocaml/xapi/xapi.ml, line 1548
    Jun 19 04:58:41 lcy2-dt72 xapi: [error||0 ||backtrace] 7/7 xapi Called from file ocaml/libs/log/debug.ml, line 250
    ```
```

**Pull request merged**